symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/installAddonAPIAction.class.php

```diff
@@ -30,60 +30,55 @@ class installAddonAPIAction extends baseAddonAction
      */
     public function execute($request)
     {
-        $addonList = $this->getAddons();
-        $data = $request->getParameterHolder()->getAll();
-        $addonId = $data['installAddonID'];
-        $addonURL = null;
-        $addonDetail = null;
-        foreach ($addonList as $addon) {
-            if ($addon['id'] == $addonId) {
-                $addonDetail = $addon;
-                $addonURL = $addon['links']['file'];
+        try {
+            $addonList = $this->getAddons();
+            $data = $request->getParameterHolder()->getAll();
+            $addonId = $data['installAddonID'];
+            $addonURL = null;
+            $addonDetail = null;
+            foreach ($addonList as $addon) {
```

Comment:
Why looping through all add-ons to get one file?

Reply:
can you suggest a better way?

Reply:
can't you call the api with add-on id ? this leads to performance issue if there are big number of add-on list